azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -0,0 +1,96 @@
+# Microsoft Azure Linux Agent
+#
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This is a placeholder policy engine class to test that the regorus
+# dependency is correctly installed.
+# pylint: disable=too-few-public-methods
+
+import pwd
+import os
+import sys
+
+from azurelinuxagent.common import logger
+from azurelinuxagent.common.version import get_distro
+from azurelinuxagent.common.utils.flexible_version import FlexibleVersion
+from azurelinuxagent.common.event import WALAEventOperation, add_event
+from azurelinuxagent.common import conf
+
+def log_policy_info(formatted_string, op=WALAEventOperation.Policy, send_event=True):
+    """Log information to console and telemetry."""
+    logger.info("[Policy] " + formatted_string)
+    if send_event:
+        add_event(op=op, message=formatted_string)
+
+
+class PolicyEngine:
```

Comment:
I think as base class it should not be singleton. when extension policy or other policy are created by inheriting this base class, these individual policies can choose to be singleton or non-singleton depending on the application of these policies. Otherwise, when all policies are sharing the same engine, which will load policies and data for their own usage, it will become very messy and error prone. For example, if it happened different rego policy or data shared same name by mistake, the rule may not be executed as expected, but very difficult to debug. some of the method (like checking whether the policy is supported or enabled) can be static or class method though.